### PR TITLE
Add Windows installer

### PR DESCRIPTION
**Pull request by ondrowan**
To try this, follow the steps from main README / README of installer. Most of the files are just copied from [innodependencyinstaller](https://github.com/domgho/innodependencyinstaller). Redistributables are contained in installer itself. I tried it on my computer and it installed everything properly, but I'd prefer if we tired it also on some clean installation of Windows to make sure everything works as intended. The only thing that is missing for now is icon, but that can be added later.

### CODE REVIEW

**Review by yanchith — COMMENTED** [review]
Seems to work on my computer. Unfortunately my computer is not clean, so we don't know for sure 🙁, we'll have to wait until I reinstall or we find a clean computer somewhere.

I didn't spend that much time on the code, but I do have a couple of general questions:

- Is `installer/Output/HURBAN Selector-1.0.exe` supposed to be checked into version control? Seems like you just changed the target directory and forgot to delete this.
- Can this be extended to add the app icon later?
- Can this be extended to create shortcuts later?
- Can this be extended to set windows file associations later?

**Comment by ondrowan** [comment]
@yanchith I've removed that installer that was in VC by accident. Also added a possibility to create desktop and Start Menu shortcut. As for icons and file associations: yes to both.

**Comment by yanchith** [comment]
@ondrowan as we discussed in person, please add a dummy icon now that we can later replace with official one.

**Comment by janper** [comment]
@ondrowan, I was asked to ask here: https://github.com/subdgtl/HURBAN-Selector/pull/114#issuecomment-575333339

**Comment by ondrowan** [comment]
@janper Yes, it is in Start menu and it is possible to add link to logs folder there.

**Comment by janper** [comment]
> @janper Yes, it is in Start menu and it is possible to add link to logs folder there.

let's do it then, please

**Comment by ondrowan** [comment]
Done and done. Adding an icon to binary requires a manual step, but it's much less awkward than alternatives I found. The only place where I couldn't get the icon to show is the upper left corner of the application window, but I'll investigate it later.

**Comment by janper** [comment]
Yeah, please, try to resolve also the issue with the window icon. Would be weird not having it.

**Comment by ondrowan** [comment]
@janper It's there. For now, it builds with icon even in debug mode. I'd like to keep it that way just to be able to spot if something's wrong right away.

File: README.md [hunk 1]
```diff
@@ -123,6 +123,13 @@ Before running on target computer, make sure "Microsoft Visual C++
 Redistributable for Visual Studio 2019" is installed (get it
 [here](https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads))
 
+### Creating a Windows installer
+
+- Install [Inno Setup](http://www.jrsoftware.org/isinfo.php),
+- `cargo build --release`,
```
Comment (yanchith) [inline README.md]: This is now missing the `--features dist`

Comment (yanchith) [reply README.md]: Hmm, maybe this should just link to the installer readme, so we don't have to double-maintain it? They are already out of sync.

**Comment by yanchith** [comment]
@ondrowan I think this is almost good to go once you resolve my comments regarding the readme and fix merge conflicts. @janper do you require anything else here?

**Review by yanchith — DISMISSED** [review]

**Comment by janper** [comment]
I'm indifferent to this.

**Review by janper — CHANGES_REQUESTED** [review]
I gave it a try. 
if I understand correctly, you first need to build the binary, then add the icon, then compile the installer? If so, please rewrite the readme accordingly.

Also (maybe some sort of my fault) but I'm unable to `cargo build --release --features dist`. Resulting with the following error:
```
Compiling hurban_selector v0.1.0 (C:\Users\daroi\Documents\HURBAN-Selector)
error[E0308]: mismatched types
  --> src\logger.rs:78:9
   |
77 |     let path = match dirs::data_local_dir() { 
   |                      ---------------------- this match expression has type `std::option::Option<std::path::PathBuf>`
78 |         Ok(appdata_local) => Path::new(&appdata_local).join("HURBAN Selector/Logs"),
   |         ^^^^^^^^^^^^^^^^^ expected enum `std::option::Option`, found enum `std::result::Result`  
   |
   = note: expected type `std::option::Option<std::path::PathBuf>`
              found type `std::result::Result<_, _>`

error[E0308]: mismatched types
  --> src\logger.rs:79:9
   |
77 |     let path = match dirs::data_local_dir() { 
   |                      ---------------------- this match expression has type `std::option::Option<std::path::PathBuf>`
78 |         Ok(appdata_local) => Path::new(&appdata_local).join("HURBAN Selector/Logs"),
79 |         Err(_) => return base_logger,
   |         ^^^^^^ expected enum `std::option::Option`, found enum `std::result::Result`
   |
   = note: expected type `std::option::Option<std::path::PathBuf>`
              found type `std::result::Result<_, _>`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `hurban_selector`.        

To learn more, run the command again with --verbose.
```

**Comment by janper** [comment]
Also I'd appreciate a link to rcedit binaries. I tried to build it from source but failed.
image

**Comment by yanchith** [comment]
> Also I'd appreciate a link to rcedit binaries. I tried to build it from source but failed.

There is a link in the link Ondro added to the readme. But I can point to it directly. You shouldn't build it from source.

**Comment by yanchith** [comment]
@janper please try again. I fixed the issues.

**Review by janper — APPROVED** [review]

**Comment by janper** [comment]
It's acceptable now. Two notes though (decide if relevant or not):
1. I didn't realize the file icon and the window icon may differ. I used a different icon for the exe accidentally. A more detailed description in the readme would be nice.
2. I know this is may be tricky but I'd suggest spending a couple of minutes searching to see if it is easy to fix: I don't mind the warning itself but the publisher should be "Subdigital"
image

**Comment by yanchith** [comment]
Damn. I merged before seeing your comment. Okay, I'll push a more detailed description of the icons to master in the evening. Do we want to use different icons?

I'll investigate if we can set the piblisher.

**Pull request merged**